examples/flask/example.py

```diff
@@ -18,3 +18,28 @@ def index():
 @picobox.pass_('magic')
 def magic(magic):
     return jsonify({'magic': magic})
+
+
+@app.before_request
+def serve_eggs_with_spam():
+    box = picobox.Box()
+
+    # on requests to /eggs, override the value of magic with 'spam'
+    if request.path == '/eggs':
+        box.put('magic', 'spam')
+
+    picobox.push(box, chain=True)
```

Comment:
(general thought) I don't have strong opinion on this yet. I'm afraid this could be not so good practice from testing POV. Just imaging, you have in tests something like:


```
with picobox.push(testbed):
    with app.test_client() as client:
        client.get('/eggs')
```

If I'm not mistaken, on `client.get('/eggs')` the `@app.before_request` would be called and will override the `testbox` you set in the test.

Maybe we should also add some sort to hook some boxes to be on top of the stack all the time (surely, not part of this pr)? What do you think?

Reply:
Discussed this offline :)